env-hooks/99.roslisp.sh

```diff
@@ -18,4 +18,4 @@ for workspace in workspaces:
 print(os.pathsep.join(paths))
 EOF
 )
-export ROSLISP_PACKAGE_DIRECTORIES="`python -c \"$PYTHON_CODE_BUILD_ROSLISP_PACKAGE_DIRECTORIES\"`"
+export ROSLISP_PACKAGE_DIRECTORIES="`python${ROS_PYTHON_VERSION} -c \"$PYTHON_CODE_BUILD_ROSLISP_PACKAGE_DIRECTORIES\"`"
```

Comment:
Is this an expected usage for `ROS_PYTHON_VERSION`? My expectation would have been to see something more like renaming this file to `99.roslisp.sh.in` and then using cmake's `PYTHON_EXECUTABLE` variable directly.

Reply:
No, since you can't assume that there is an executable named `python2` / `python3`. E.g. in a venv or on Windows the executable will just be `python`.

@sloretz FYI - this probably needs fixing.

Reply:
> No, since you can't assume that there is an executable named python2 / python3. E.g. in a venv or on Windows the executable will just be python.
> 
> @sloretz FYI - this probably needs fixing.

I think this is fine as is. I'm assuming this environment hook did not work on windows before this PR.

> This uses `python2` or `python3` instead of `python` with the version being chosen using the environment variable `ROS_PYTHON_VERSION`. This should work fine on any system compatible with [PEP394](https://www.python.org/dev/peps/pep-0394/), which seems reasonable given this is an `sh` shell environment hook.

Generating the script from CMake could remove the PEP 394 assumption. It seems like it would add an assumption that the path to python on the machine building the package is the same path to python on the machine the package is installed on, which might be ok but may make it more difficult to use a `venv` with ROS.

Reply:
> it would add an assumption that the path to python on the machine building the package is the same path to python on the machine the package is installed on

This assumption is already embedded into every script installed using `catkin_install_python`.